File: packages/pluto-py/pluto_client/reactapp.py
from typing import Optional
from dataclasses import dataclass
from .website import Website, WebsiteOptions, IWebsiteClient


@dataclass
class ReactAppOptions(WebsiteOptions):
    """
    The options for instantiating a resource class.
    """

    buildDir: Optional[str] = "build"
    """
    The directory path to the React app build output, relative to the project path.
    """

    buildCommand: Optional[str] = "npm run build"
    """
    The command for building the React app.
    """


class ReactApp(Website):
    fqn = "@plutolang/pluto.ReactApp"

    def __init__(
        self,
        project_path: str,
        name: Optional[str] = None,
        options: Optional[ReactAppOptions] = None,
    ):
        self._client = Website(project_path, name, options)

File: packages/pluto/src/reactapp.ts
import { IResource } from "@plutolang/base";
import { IWebsiteClient, Website, WebsiteOptions } from "./website";

/**
 * The options for instantiating a resource class.
 */
export interface ReactAppOptions extends WebsiteOptions {
  /**
   * The directory path to the React app build output, relative to the project path.
   * @default "build"
   */
  buildDir?: string;
  /**
   * The command for building the React app.
   * @default "npm run build"
   */
  buildCommand?: string;
}

/**
 * The ReactApp resource type.
 */
export class ReactApp implements IResource {
  /**
   * The fully qualified name of the resource type.
   */
  public static fqn = "@plutolang/pluto.ReactApp";

  constructor(projectPath: string, name?: string, options?: ReactAppOptions) {
    name;
    projectPath;
    options;
    throw new Error(
      "Cannot instantiate this class, instead of its subclass depending on the target runtime."
    );
  }

  public static buildClient(
    projectPath: string,
    name?: string,
    options?: ReactAppOptions
  ): IWebsiteClient {
    return Website.buildClient(projectPath, name, options);
  }
}

export interface ReactApp extends Website {}
